internal/js-dependency-scanner/src/main/java/org/wso2/security/tools/scanner/dependency/js/issuecreator/JIRAIssueCreator.java

```diff
@@ -0,0 +1,246 @@
+                //Get the index of the last occurrence of the character ('-') in product name.
+                int index = entry.getKey().lastIndexOf("-");
+                String summary = entry.getKey().substring(0, index) + ":" + ticketSubject;
+                String issueKey = checkForIssueExistence(auth, summary, projectKey);
+                if (StringUtils.isBlank(issueKey)) {
+                    createJIRAIssue(entry.getKey(), fileMapper.get(entry.getKey()), auth, summary);
+                } else {
+                    try {
+                        jira.invokePostComment(auth, getEndPointURL() +
+                                IssueCreatorConstants.JIRA_REST_API_URL +
+                                "/" + issueKey + "/comment", createComment(entry.getKey()));
+                    } catch (AuthenticationException e) {
+                        throw new IssueCreatorException("Failed to add comment for attached file in jira ticket." +
+                                entry.getKey().substring(0, index), e);
+                    }
+                    log.info("[JS_SEC_DAILY_SCAN] JIRA Comment added : " + createComment(entry.getKey()));
+                    jira.invokePutMethodWithFile(auth, getEndPointURL() +
+                            IssueCreatorConstants.JIRA_REST_API_URL + "/" +
+                            issueKey + "/attachments", fileMapper.get(entry.getKey()));
+                }
+            }
+        }
+    }
+
+    /**
+     * Create JIRA Issue.
+     *
+     * @param product  product name
+     * @param filePath File path where the generated report is located. It will attached with ticket.
+     * @throws IssueCreatorException exception occurred while creating issue ticket.
+     */
+    private void createJIRAIssue(String product, String
+            filePath, String auth, String summary) throws IssueCreatorException {
+        String issue;
+        String assignee = null;
+        if (product.contains(JSScannerConstants.AM)) {
+            assignee = this.getAssigneeMapper().get(JSScannerConstants.AM);
+        } else if (product.contains(JSScannerConstants.INTEGRATION)) {
+            assignee = this.getAssigneeMapper().get(JSScannerConstants.INTEGRATION);
+        } else if (product.contains(JSScannerConstants.STREAMPROCESSOR)) {
+            assignee = this.getAssigneeMapper().get(JSScannerConstants.STREAMPROCESSOR);
+        } else if (product.contains(JSScannerConstants.IDENTITYSERVER)) {
+            assignee = this.getAssigneeMapper().get(JSScannerConstants.IDENTITYSERVER);
+        } else if (product.contains(JSScannerConstants.OB)) {
+            assignee = this.getAssigneeMapper().get(JSScannerConstants.OB);
+        }
+
+        if (assignee == null) {
+            assignee = IssueCreatorConstants.DEFAULT_ASSIGNEE;
+        }
+
+        String jiraBaseUrl = getEndPointURL();
+        String issueToBeCreated = createNewTicket(product, assignee, summary);
```

Comment:
createNewTicket() name is misleading. It implies creating an actual ticket in the JIRA via the API. But what the code is doing is building the ticket data.